core/src/test/java/org/apache/accumulo/core/data/NamespaceIdTest.java

```diff
@@ -79,12 +79,13 @@ public void testCacheIncreasesAndDecreasesAfterGC() {
     assertEquals(namespaceString, nsId.canonical());
 
     // create a bunch more and throw them away
-    for (int i = 0; i < 999; i++) {
-      NamespaceId.of(new String("namespace" + i));
+    long preGCSize = 0;
+    int i = 0;
+    while ((preGCSize = NamespaceId.cache.asMap().entrySet().stream().count()) < 100) {
```

Comment:
```suggestion
    while ((preGCSize = NamespaceId.cache.asMap().entrySet().size()) < 100) {
```

Reply:
We can't use `size()` (either on the `Map` or its `entrySet()`), because that is only an approximation when working with Guava Cache objects. We need the actual count, which we can get by iterating over the entry set. See https://guava.dev/releases/21.0/api/docs/com/google/common/cache/Cache.html#size--

Reply:
Ah thank you.  I was racking my brain trying to figure out the difference between the two.